src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -163,10 +163,25 @@ void ComponentFiller::addVariables(Optimisation::LinearProblemApi::ILinearProble
                                                                data);
 
     Expressions::Visitors::EvalVisitor evaluator(evaluationContext, ctx);
+    auto valueOrDefault = [&evaluator](const auto& node, double defaultValue)
+    {
+        if (node.Empty())
+        {
+            return Expressions::Visitors::EvaluationResult(defaultValue);
+        }
+        return evaluator.dispatch(node.RootNode());
+    };
     for (const auto& variable: component_.getModel()->Variables() | std::views::values)
     {
-        const auto& lb = evaluator.dispatch(variable.LowerBound().RootNode());
-        const auto& ub = evaluator.dispatch(variable.UpperBound().RootNode());
+        namespace SM = ModelerStudy::SystemModel;
+        const auto& lb = valueOrDefault(variable.LowerBound(),
```

Comment:
is it possible to do this only once, when creating the model in the library ? I feel like if we do it here, we might have an avoidable overhead during runtime, since the same operations will be done for every component

Reply:
modelConverter.cpp:convertVariables would be the best candidate but at this point we only have expression so it depends if we already support -inf/+inf expressions

Reply:
https://gopro-tickets.rte-france.com/browse/ANT-3255